### PR TITLE
Added flag to skip checks to save time on submission end.  Implemented here for RCDS

### LINKED ISSUE #323: Implement a flag that skips RCDS publish check at submission
Copied from ServiceNow ticket RITM1654874 by @hgreenlee:

This is a feature request to jobsub_lite developers.  I would like there to be a jobsub_submit option such that when jobsub_submit is invoked with option "-f dropbox://", jobsub_submit does not wait for RCDS upload before exiting and printing the submitted job id.  I don't care if waiting for RCDS stays as the default.  Obviously, this implies that the batch job wrapper has to be smart enough to wait for the RCDS upload (if it doesn't do that already).  The reasoning behind this request is that there are use cases where wall time of job submission is more valuable than wall time in batch jobs, even potentially at the expense of some batch job efficiency metric.

Extra notes by @shreyb:

I discussed this with @marcmengel, and we agreed that this was a useful and reasonable feature to add.  

However, since this does introduce risk that the job may fail due to the failure of RCDS to publish the tarball, this change MUST include a warning message of some sort to the user if they use this proposed flag that warns them of the risk, and instructs them to resubmit with the flag removed if they do not want to accept the risk.

### PR DESCRIPTION
**Pull request by shreyb**
Closes #323

### CODE REVIEW

**Comment by shreyb** [comment]
I have two unit tests that are failing.  Will check those, and run a couple more tests, before moving this to "ready".

**Comment by shreyb** [comment]
Rebased after #325 merged.  Now all unit tests, including new ones, pass.  Will move this to "Ready" as soon as I document CLI tests I did here.

**Comment by shreyb** [comment]
Note to @marcmengel or any other reviewer:  You probably will want to hide whitespace for this review - `tarfiles.py` has a large chunk of code that's just indented, but it's hard to tell with the unified or side-by-side diff with whitespace.

File: lib/tarfiles.py
Comment (marcmengel) [reply lib/tarfiles.py:280]: So old jobsub uses a shell function in the wrapper:
https://cdcvs.fnal.gov/redmine/projects/jobsub/repository/revisions/master/entry/lib/groupsettings/JobUtils.py#L303
I note that it doesn't actually call the web api, which is probably good, it just polls the possible locations, given the cid. 
It might be cleaner to put a simpler version in the simple.sh wrapper.

Comment (marcmengel) [reply lib/tarfiles.py:280]: So it old jobsub ends up looking at, i.e. https://rcds01.fnal.gov/pubapi/config and getting back
`repos:fifeuser1.opensciencegrid.org,fifeuser2.opensciencegrid.org,fifeuser3.opensciencegrid.org,fifeuser4.opensciencegrid.org`
and drops those in  the wrapper and  has a for loop over each of those  looking in /cvms/{repo}/sw/{cid} . 

So to emulate this much more simply if we're punting on the publish check, we should:
* lookup the config entry and get the repo list in repolist
* set location to a glob string like: `f'/cvmfs/\{{",".join(repolist)}\}/sw/{digest}'`
* make sure our loop in simple.sh will work with that glob pattern.

Comment (shreyb) [reply lib/tarfiles.py:280]: Working on a rewrite of this PR that includes all of this.  I realized that a lot of what I did with the skip_checks.py can be cleanly done with an Enum, so I'm redoing it.  I expect it'll be ready by the end of the week.

Comment (shreyb) [reply lib/tarfiles.py:280]: Almost done with this.  Just need to run some test.  I'm going to put this in Draft, and then mark it Ready when it's....Ready.

**Review by marcmengel — APPROVED** [review]
Looks good!

**Comment by shreyb** [comment]
Thanks @marcmengel  - I'm going to dismiss your review only because I have one or two more commits to push here.  But I'm glad it looks good so far! 

There were a couple bugs with the way it is now that I fixed with subsequent tests.

**Comment by shreyb** [comment]
Done.  How does that look?

**Comment by marcmengel** [comment]
That looks good.   One other thing though, I ran test jobs, and one of them (regular, non-deferred-check RCDS) did all 10 checks and still didn't find the file and then failed:

Looking for directory /cvmfs/fifeuser2.opensciencegrid.org/sw/dune/30c08967c4046571dbf76fcde600018061951b3474a8c97
d6261d35615177ce5 on RCDS.  Try 10 of 10
Max retries 10 exceeded to find /cvmfs/fifeuser2.opensciencegrid.org/sw/dune/30c08967c4046571dbf76fcde600018061951
b3474a8c97d6261d35615177ce5.  Job may fail.

Do you think we need the limit to be 15 or 20 in the single case?

**Comment by marcmengel** [comment]
Actually, thinking about it, 300 sec should be plenty long for a file that's been uploaded; if the cvmfs client is not caught up, there may be other problems as well...

**Comment by shreyb** [comment]
No - I actually agree with your earlier comment.  Right now, we calculate the number of total tries by taking the length of the choices list (so it'd be 4 at the moment) and multiplying it by the minimum tries per choice (10 currently).  So we allow for 40 retries if we don't know the path, and only 10 if we do.  There's also a cap for how many tries we can have overall for each tarfile, which is also 40.  

We could bump up that minimum tries per choice to 15 or 20, so that if there's 1 or 2 choices, we'd try each one 15 or 20 times, but if there were more choices than that (as we currently have in production), the maximum retries would curtail that number at 40 total tries.

So in the current production state, if we know the path, we'd try it 15 or 20 times (depending on what that number is); and if we don't, we'd try each 10 times on average (assuming our randomizer is perfect).

It's an easy enough change - what do you think?

**Comment by marcmengel** [comment]
I think we could raise the minimum a bit.    I mean, we're checking every 30 seconds so we're not really wasting more than 30 seconds of slot time; it's more a question of how long do we think it might take if it's working.  Certainly an hour is too long; but going up to 10 minutes isn't bad.

**Pull request merged**
